src/PoissonSolvers/FFTOpenPoissonSolver.hpp

```diff
@@ -1141,6 +1203,124 @@ namespace ippl {
+                    double v = kg * hs_m[2];
+
+                    double s = (t * t) + (u * u) + (v * v);
+                    s        = std::sqrt(s);
+
+                    view_g2n1(i, j, k) =
+                        -2 * (std::sin(0.5 * L_sum * s) / s) * (std::sin(0.5 * L_sum * s) / s);
+
+                    // if (0,0,0), assign L^2/2 (analytical limit of sinc)
+                    if ((ig == 0 && jg == 0 && kg == 0)) {
+                        view_g2n1(i, j, k) = -L_sum * L_sum * 0.5;
+                    }
+                });
+
+            // start a timer
+            static IpplTimings::TimerRef fft4 = IpplTimings::getTimer("FFT: Precomputation");
+            IpplTimings::startTimer(fft4);
+
+            // inverse DCT transform of 2N+1 green's function for the precomputation
+            fft2n1_m->transform(BACKWARD, grn2n1_m);
+
+            IpplTimings::stopTimer(fft4);
+
+            // Restrict transformed grn2n1_m to 2N domain after precomputation step
+
+            // get the field data first
+            typename Field_t::view_type view = grn_mr.getView();
+            const int nghost                 = grn_mr.getNghost();
+            const auto& ldom                 = layout2_m->getLocalNDIndex();
+
+            // start a timer
+            static IpplTimings::TimerRef ifftshift = IpplTimings::getTimer("Vico shift loop");
+            IpplTimings::startTimer(ifftshift);
+
+            // get number of ranks to see if need communication
+            int ranks = ippl::Comm->size();
+
+            // range type for Kokkos loop
+            using mdrange_type = Kokkos::MDRangePolicy<Kokkos::Rank<3>>;
+
+            if (ranks > 1) {
+                communicateVico(size, view_g2n1, ldom_g2n1, nghost_g2n1, view, ldom, nghost);
+            } else {
+                // restrict the green's function to a (2N)^3 grid from the (2N+1)^3 grid
+                Kokkos::parallel_for(
+                    "Restrict domain of Green's function from 2N+1 to 2N",
+                    mdrange_type({nghost, nghost, nghost}, {view.extent(0) - nghost - size[0],
+                                                            view.extent(1) - nghost - size[1],
+                                                            view.extent(2) - nghost - size[2]}),
+                    KOKKOS_LAMBDA(const int i, const int j, const int k) {
+                        // go from local indices to global
+                        const int ig = i + ldom[0].first() - nghost;
+                        const int jg = j + ldom[1].first() - nghost;
+                        const int kg = k + ldom[2].first() - nghost;
+
+                        const int ig2 = i + ldom_g2n1[0].first() - nghost_g2n1;
+                        const int jg2 = j + ldom_g2n1[1].first() - nghost_g2n1;
+                        const int kg2 = k + ldom_g2n1[2].first() - nghost_g2n1;
+
+                        if ((ig == ig2) && (jg == jg2) && (kg == kg2)) {
```

Comment:
Would it be possible to remove this if too by using a boolean? This is still there in the master too inside the ifftshift loop of normal Vico, so it would have to be changed both here (DCT_VICO) and there (normal VICO - [line 1186](https://github.com/IPPL-framework/ippl/blob/051bc8abdef69c2d68fcc447c53b1c66727e3a45/src/PoissonSolvers/FFTOpenPoissonSolver.hpp#L1186)).